fvm/evm/emulator/emulator.go

```diff
@@ -381,12 +370,16 @@ func (proc *procedure) runDirect(
 	msg *gethCore.Message,
 	txHash gethCommon.Hash,
 	txIndex uint,
-	txType uint8,
 ) (*types.Result, error) {
 	// set the nonce for the message (needed for some opeartions like deployment)
 	msg.Nonce = proc.state.GetNonce(msg.From)
 	proc.evm.TxContext.Origin = msg.From
-	return proc.run(msg, txHash, txIndex, types.DirectCallTxType)
+	res, err := proc.run(msg, txHash, txIndex, types.DirectCallTxType)
+	if err != nil {
+		return res, err
```

Comment:
Do we want to return `res` here? Shouldn't it be `return nil, err` ? Or is there a reason for it?

Reply:
fixed it to return res